#162727410 Setup empty react project

daniellamarr opened the pull request:
#### What does this PR do?
- This PR initializes an empty react project
#### Description of Task to be completed?
- The application should run on webpack
- Babel should be installed for transpiling ES6+
#### How should this be manually tested?
- Pull this branch off the repository
- Run `npm start` on your local machine
#### Any background context you want to provide?
- The app would run on port `5050` on a local machine
#### What are the relevant pivotal tracker stories?
[#162727410](https://www.pivotaltracker.com/story/show/162727410)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

daniellamarr committed:
Chore(Setup empty react project):
Setup an empty react project with webpack
[Delivers #162727410]

abejide001 commented on webpack.config.js:
```diff
@@ -0,0 +1,30 @@
+  plugins: [
+    new webpack.HotModuleReplacementPlugin()
+  ],
+  devServer: {
```
please add this property to devServer `historyApiFallback: true`, because without this you will get an error on page reload.

daniellamarr replied:
Done

abejide001 reviewed (changes requested):
I left a comment for you

andela-dbamidele commented on webpack.config.js:
```diff
@@ -0,0 +1,30 @@
+const webpack = require('webpack');
```
You should have a separate config for dev and prod. I'll suggest you create 3 files, one will be the base file that will contain shared settings between dev and prod and the other 2 will contain the environment specific config.

One example is devServer, you only need it for development and not on production. Also, you might need to add some optimization plugins specifically for production.

daniellamarr replied:
I have done that but I'm not sure of the optimization plugins for production

andela-dbamidele commented on dist/index.html:
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
```
What's the purpose of the dist folder and what other files do you want to put there.

daniellamarr replied:
The dist folder would hold production-ready code.
Files like CSS and JS would go in eventually but for now, only "index.html" would be there.

abejide001 replied:
@daniellamarr, I think the CSS files should be in an asset folder

daniellamarr replied:
I just realized the `dist` folder should not be created by me but by the webpack config I already set up. The dist folder is created when a build is ran on the app and only if I specify that the target folder should be `dist`

daniellamarr committed:
Chore(Setup empty react project)
Refactor webpack config
Add production and development environment
[Finishes #162727410]

abejide001 commented on package.json:
```diff
@@ -4,7 +4,9 @@
   "scripts": {
-    "start": "webpack-dev-server --config ./webpack.config.js --mode development",
+    "start": "npm run build",
+    "start:dev": "webpack-dev-server --config ./webpack/webpack.dev.js --mode development",
```
I don't think you still need to specify `--mode development` in the `start:dev`, since you have created `webpack/webpack.dev.js` and `mode` property is specified there already.

daniellamarr replied:
Done

daniellamarr committed:
Chore(Create empty react project):
Remove dist folder
Add index.html, App.jsx, App.css files to src folder
Refactor webpack.common.js file
[Finieshes #162727410]

daniellamarr committed:
Chore(Create empty react project):
Add new line to .babelrc file
[Finishes #162727410]

andela-dbamidele commented on webpack/webpack.common.js:
```diff
@@ -0,0 +1,37 @@
+      template: './src/index.html',
+      filename: './index.html'
+    }),
+    new MiniCSSExtractPlugin({
```
Is this part of your story?

andela-dbamidele commented on webpack/webpack.dev.js:
```diff
@@ -0,0 +1,6 @@
+const common = require('./webpack.common.js');
+
+module.exports = {
```
There's a package called `webpack-merge`. Please research that and make use of it.

andela-dbamidele commented on webpack/webpack.common.js:
```diff
@@ -0,0 +1,37 @@
+      chunkFilename: "[id].css"
+    })
+  ],
+  devServer: {
```
This should be in the dev config.

andela-dbamidele commented on webpack/webpack.prod.js:
```diff
@@ -0,0 +1,6 @@
+const common = require('./webpack.common.js');
+
+module.exports = {
```
Same here.

andela-dbamidele commented on webpack/webpack.common.js:
```diff
@@ -0,0 +1,37 @@
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const MiniCSSExtractPlugin = require('mini-css-extract-plugin');
```
What's this plugin doing?

daniellamarr committed:
Chore(Setup empty react project):
Refactor webpack files
Removed App.css
Remove css-loader, sass-loader, style-loader
Add minify package
[Finishes #162727410]
